src/main/java/com/hillel/items_exchange/controller/AdvertisementController.java

```diff
@@ -114,6 +116,29 @@
         return ResponseEntity.unprocessableEntity().body(HttpStatus.CONFLICT);
     }
 
+    @PostMapping("/setDefaultImage/{advertisementId}/{imageId}")
+    public ResponseEntity<HttpStatus> setAdvertisementDefaultImage(@PathVariable Long advertisementId,
+                                                                   @PathVariable Long imageId,
+                                                                   Principal principal) {
+        User owner = getUser(principal.getName());
+        final Advertisement advertisement = owner.getAdvertisements().stream()
```

Comment:
Move this logic to a new method because now you break SOLID ``Single responsibility`` principal. And better to move it to the service layer.